cmd/zpool/zpool_iter.c

```diff
@@ -332,9 +332,10 @@ vdev_run_cmd_thread(void *cb_cmd_data)
 	char cmd[_POSIX_ARG_MAX];
 
 	/* Set our VDEV_PATH and VDEV_UPATH env vars and run command */
-	if (snprintf(cmd, sizeof (cmd), "VDEV_PATH=%s && VDEV_UPATH=%s && %s",
-	    data->path, data->upath ? data->upath : "\"\"", data->cmd) >=
-	    sizeof (cmd)) {
+	if (snprintf(cmd, sizeof (cmd), "VDEV_PATH=%s && VDEV_UPATH=\"%s\" && "
+	    "VDEV_ENC_SYSFS_PATH=\"%s\" && %s", data->path,
+	    data->upath, data->vdev_enc_sysfs_path,
+	    data->cmd) >= sizeof (cmd)) {
 		/* Our string was truncated */
```

Comment:
Will `snprintf` and `strdup` (below) always handle a NULL `vdev_enc_sysfs_path` value?